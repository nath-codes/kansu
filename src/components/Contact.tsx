import { rem } from 'polished'
import React, { memo } from 'react'
import styled from 'styled-components'

import { Container } from './Container'
import { Heading } from './Heading'
import { Section } from './Section'
import { Text } from './Text'

import {
  ARIA_EMAIL,
  ARIA_GITHUB,
  ARIA_LINKEDIN,
  URL_EMAIL,
  URL_GITHUB,
  URL_LINKEDIN,
} from '../constants'

const StyledText = styled(Text)`
  margin-bottom: ${rem(24)};

  @media only screen and (min-width: ${(props) => props.theme.breakpoints.m}) {
    margin-bottom: ${rem(48)};
  }
`

const _Contact = () => {
  return (
    <Section>
      <Heading>Contact</Heading>
      <Container>
        <StyledText>
          <p>
            I am available for fully-remote contracts, or for hybrid-based
            contracts in the UK. If you would like to get in touch to discuss
            any opportunities, please contact me via{' '}
            <a href={URL_EMAIL} aria-label={ARIA_EMAIL}>
              email
            </a>
            .
          </p>

          <p>
            Alternatively, you can contact me on{' '}
            <a href={URL_LINKEDIN} aria-label={ARIA_LINKEDIN}>
              LinkedIn
            </a>{' '}
            or view examples of my work via my{' '}
            <a href={URL_GITHUB} aria-label={ARIA_GITHUB}>
              Github
            </a>
            .
          </p>
        </StyledText>
      </Container>
    </Section>
  )
}

export const Contact = memo(_Contact)
