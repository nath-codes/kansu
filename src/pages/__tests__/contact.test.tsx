import 'jest-styled-components'
import React from 'react'
import { renderWithTheme } from '../../components/__tests__/helpers'
import Contact from '../contact'

jest.mock('../../components/seo')

describe('Contact', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Contact />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
