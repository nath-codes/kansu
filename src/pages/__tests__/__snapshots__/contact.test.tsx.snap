// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `
Array [
  <div>
    Contact
  </div>,
  .c6 {
  background: #ffffff;
  border: solid 0.375rem #ffffff;
  color: #202f3c;
  font-family: "Roboto",sans-serif;
  font-size: 1.125rem;
  -webkit-letter-spacing: 2.3px;
  -moz-letter-spacing: 2.3px;
  -ms-letter-spacing: 2.3px;
  letter-spacing: 2.3px;
  line-height: 1;
  padding: 0.5625rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: background 0.3s,color 0.3s;
  transition: background 0.3s,color 0.3s;
}

.c6:hover {
  background: none;
  color: #ffffff;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 3rem;
}

.c1 {
  color: #ffffff;
  font-family: "Roboto",sans-serif;
  font-size: 2.25rem;
  left: 0;
  -webkit-letter-spacing: 2.3px;
  -moz-letter-spacing: 2.3px;
  -ms-letter-spacing: 2.3px;
  letter-spacing: 2.3px;
  line-height: 1;
  margin: 0;
  padding-right: 0.75rem;
  position: absolute;
  text-transform: uppercase;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c1:after {
  color: #ffffff;
  content: 'â– ';
  bottom: 0.25rem;
  font-size: 0.75rem;
  position: absolute;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  position: relative;
}

.c4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-size: 1.125rem;
  line-height: 1.25;
}

.c4 a {
  color: #49555e;
  position: relative;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c4 a:hover {
  color: #ffffff;
}

.c4 p {
  margin-bottom: 1.5rem;
}

.c4 p:last-child {
  margin-bottom: 0;
}

.c3 {
  margin-bottom: 3rem;
}

.c5 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media only screen and (min-width:768px) {
  .c2 {
    max-width: 37.5rem;
  }
}

@media only screen and (min-width:980px) {
  .c2 {
    max-width: 40.5rem;
  }
}

@media only screen and (min-width:980px) {
  .c1:after {
    bottom: 0;
    font-size: 1.5rem;
  }
}

@media only screen and (min-width:768px) {
  .c4 {
    line-height: 1.3333333333;
  }
}

@media only screen and (min-width:980px) {
  .c4 {
    margin-top: 3rem;
  }
}

<section
    className="c0"
  >
    <h2
      className="c1"
      style={
        Object {
          "marginTop": "-1.5rem",
          "opacity": 0,
          "top": "0",
        }
      }
    >
      Contact
    </h2>
    <div
      className="c2"
    >
      <div
        className="c3 c4"
        style={
          Object {
            "opacity": 0,
            "transform": "translateY(1.5rem)",
          }
        }
      >
        <p>
          I am available for contract work in Bristol, London and the South-West of the UK. If you would like to get in touch to discuss any opportunities, click the button below to drop me an email.
        </p>
        <p>
          Alternativly, view my projects on
           
          <a
            aria-label="View my projects on Github"
            href="https://github.com/nathan-kansu"
          >
            Github
          </a>
           
          or follow me on
           
          <a
            aria-label="Follow me on Instagram"
            href="https://www.instagram.com/nathan.kansu"
          >
            Instagram
          </a>
          .
        </p>
      </div>
      <a
        className="c5 c6"
        href="mailto:hello@kansu.co"
        style={
          Object {
            "opacity": 0,
            "transform": "translateY(-1.5rem)",
          }
        }
      >
        Contact
      </a>
    </div>
  </section>,
]
`;
